pkg/apis/v1alpha1/ipaddressallocation_types.go

```diff
@@ -0,0 +1,62 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	VisibilityExternal = "External"
+	VisibilityPrivate  = "Private"
+	VisibilityProject  = "Project"
+)
+
+// +genclient
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
+
+// IPAddressAllocation is the Schema for the ipallocation API.
+// +kubebuilder:printcolumn:name="IPAddressBlockVisibility",type=string,JSONPath=`.spec.ip_address_block_visibility`,description="IPAddressBlockVisibility of IPAddressAllocation"
+// +kubebuilder:printcolumn:name="CIDR",type=string,JSONPath=`.status..cidr`,description="CIDRs for the IPAddressAllocation"
+type IPAddressAllocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   IPAddressAllocationSpec   `json:"spec"`
+	Status IPAddressAllocationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// IPAddressAllocationList contains a list of IPAddressAllocation.
+type IPAddressAllocationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IPAddressAllocation `json:"items"`
+}
+
+// IPAddressAllocationSpec defines the desired state of IPAddressAllocation.
+type IPAddressAllocationSpec struct {
+	// IPAddressBlockVisibility specified the visibility of the IPBlocks to allocate IP addresses. Can be External, Private or Project.
+	// +kubebuilder:validation:Enum=External;Private;Project
+	// +optional
+	IPAddressBlockVisibility string `json:"ip_address_block_visibility,omitempty"`
+	// IPBlock specifies the IPBlock to allocate IP addresses from, if not specified, it will be allocated from the IPBlocks of the VPC.
+	IPBlock string `json:"ip_block,omitempty"`
```

Comment:
for WCP case, we do not need ip_block params.

Reply:
Humm.. this may be true. Then probably remove it for now.

Reply:
ok